homeassistant/components/trafikverket_train/config_flow.py

```diff
@@ -83,19 +132,15 @@ async def async_step_reauth_confirm(
             api_key = user_input[CONF_API_KEY]
 
             assert self.entry is not None
-            try:
-                await self.validate_input(
-                    api_key, self.entry.data[CONF_FROM], self.entry.data[CONF_TO]
-                )
-            except InvalidAuthentication:
-                errors["base"] = "invalid_auth"
-            except NoTrainStationFound:
-                errors["base"] = "invalid_station"
-            except MultipleTrainStationsFound:
-                errors["base"] = "more_stations"
-            except Exception:  # pylint: disable=broad-exception-caught
-                errors["base"] = "cannot_connect"
-            else:
+            errors = await validate_input(
+                self.hass,
+                api_key,
+                self.entry.data[CONF_FROM],
+                self.entry.data[CONF_TO],
+                dt_util.parse_time(self.entry.data[CONF_TIME]),
```

Comment:
maybe move the parsing of time to the validate function? It would clean it up I think